### PR TITLE
[13.0][IMP] base_partner_sequence: improve write performance

### PR DESCRIPTION
**Pull request by JordiMForgeFlow**
Prevent forcing to do record by record write always and for all the records, and only do it when needed.

CC @ForgeFlow

### CODE REVIEW

File: base_partner_sequence/models/partner.py [hunk 1]
```diff
@@ -28,16 +28,16 @@ def copy(self, default=None):
...
+            partners_needing_ref = self.filtered(
+                lambda p: p._needs_ref(vals=vals) and not p.ref
+            )
+            for partner in partners_needing_ref:
+                partner_vals = vals.copy()
                 partner_vals["ref"] = partner._get_next_ref(vals=partner_vals)
-            super(ResPartner, partner).write(partner_vals)
-        return True
+                super(ResPartner, partner).write(partner_vals)
+        return super(ResPartner, self - partners_needing_ref).write(vals)
```
Comment (MiquelRForgeFlow) [inline base_partner_sequence/models/partner.py:40]:
```suggestion
        return super().write(vals)
```

**Review by MiquelRForgeFlow — APPROVED** [review]
Needs forward ports.

**Review by HviorForgeFlow — APPROVED** [review]

**Comment by JordiMForgeFlow** [comment]
Hi @pedrobaeza 

Could you review this small improvement? :)

**Pull request merged**